### PR TITLE
Update ctags.readme.txt

### PR DESCRIPTION
**Pull request by ssiccha**
- Clarify ctags.readme.txt

Clarify that the content of ctags_for_gap goes into a file `~/.ctags`.
The previous explanation could also be read as "copy the file
ctags_for_gap into a directory ~/.ctags".

- Update ctags.readme.txt

Add instructions for usage with Universal Ctags.

Add a link to instructions on how to create tags via git hooks.

# Description

## Further details

I misread the instruction in `etc/ctags.readme.txt`. While that was my mistake, I think the formulation I suggest in the first commit makes it more clear that `~/.ctags` should be a file and not a directory.

In the second commit I update the instructions such that they also apply to Universal Ctags.

# Checklist for pull request reviewers

- [ ] proper formatting

If your code contains kernel C code, run `clang-format` on it; the 
simplest way is to use `git clang-format`, e.g. like this (don't
forget to commit the resulting changes):

    git clang-format $(git merge-base HEAD master)

- [ ] usage of relevant labels

   1. either `release notes: not needed` or `release notes: to be added`
   2. at least one of the labels `bug` or `enhancement` or `new feature`
   3. for changes meant to be backported to `stable-4.X` add the `backport-to-4.X` label
   4. consider adding any of the labels `build system`, `documentation`, `kernel`, `library`, `tests`

- [ ] runnable tests
- [ ] lines changed in commits are sufficiently covered by the tests
- [ ] adequate pull request title
- [ ] well formulated text for release notes
- [ ] relevant documentation updates
- [ ] sensible comments in the code

### CODE REVIEW

File: etc/ctags.readme.txt [hunk 1]
```diff
@@ -1,9 +1,25 @@
-Copy the file ctags_for_gap to ~/.ctags and then run
+Run
```
Comment (fingolfin) [inline etc/ctags.readme.txt:1]: Maybe add an initial sentence/paragraph to explain what this is all about?

File: etc/ctags.readme.txt [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+  ctags --version
+to see whether you are running Exuberant Ctags or Universal Ctags.
+Append the content of the file ctags_for_gap to the file
+  ~/.ctags
```
Comment (fingolfin) [inline etc/ctags.readme.txt:5]: Note that none of this is necessary if one just wants to use ctags in GAP, at least with excuberant ctags: we already have a .ctags file (well, symlink) in the gap root dir!

File: etc/ctags.readme.txt [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
+or
+  $HOME/.config/ctags/gap.ctags
+if $XDG_CONFIG_HOME is not defined.
+Then run
   ctags --sort=no lib/*.g lib/*.gi lib/*.gd src/*
-in the GAP installation directory. Start your editor
-in that directory and then do (for vim e.g.):
+in the GAP installation directory. This creates a file
+  tags
+Start your editor in that directory and then do (for vim e.g.):
```
Comment (fingolfin) [inline etc/ctags.readme.txt:16]: This is phrased weirdly. Maybe:
```suggestion
Many editors support ctags either natively or via a plugin (e.g. vim, emacs, BBEdit, ...). For example, to use it in vim, start it in that directory and then do:
```

File: etc/ctags.readme.txt (hunk 2 above)
Comment (ssiccha) [reply etc/ctags.readme.txt:5]: Which of course does not work with Universal Ctags *sigh*...
Anyways, I'll add a comment at the top of the file.

**Comment by ssiccha** [comment]
@fingolfin Updated.

**Review by wilfwilson — APPROVED** [review]
Looks fine to me, not that I'm an expert on Ctags.

File: etc/ctags.readme.txt [hunk 4]
```diff
@@ -1,9 +1,44 @@
...
+  make tags
+If you want to use ctags in another GAP project or have Universal Ctags
+installed, read on.
+
+Run
+  ctags --version
+to see whether you are running Exuberant Ctags or Universal Ctags.
+Append the content of the file ctags_for_gap to the file
+  ~/.ctags
+if you are running Exuberant Ctags. If you are running Universal Ctags then
```
Comment (ruthhoffmann) [inline etc/ctags.readme.txt:20]: Make the two cases clearer here. Maybe use paragraphs?

**Review by ruthhoffmann — COMMENTED** [review]
I appreciate the context that has been added based on @fingolfin's comment. I'm not an expert either but except for some minor formatting this makes sense and I could use Ctags myself!

File: etc/ctags.readme.txt [hunk 5]
```diff
@@ -1,9 +1,44 @@
...
-in the GAP installation directory. Start your editor
-in that directory and then do (for vim e.g.):
+This file describes how to set up Ctags to work with GAP source files.
+Ctags is a tool that can be used to generate an index file of names found in
+source files. This index file can then be used by editors to jump to the
+definition of any indexed name.
+
+Note that if you have Exuberant Ctags installed (if in doubt, run `ctags --version`)
+and only want to generate tags for the gap-system/gap repository, then you do
+not need set up anything. The symlink `.ctags` in the root folder already
```
Comment (fingolfin) [inline etc/ctags.readme.txt:8]:
```suggestion
not need set up anything. The symlink `.ctags` in the GAP root folder already
```

File: etc/ctags.readme.txt (hunk 4 above)
Comment (fingolfin) [reply etc/ctags.readme.txt:20]: Bonus idea: convert the file to markdown and call it `README.ctags.md`; then use four space indent and surrounding empty lines to turn this into code blocks.

But of course this goes way beyond the original scope of this PR. so please do not feel compelled to do this unless you really want to.

File: etc/ctags.readme.txt [hunk 6]
```diff
@@ -1,9 +1,44 @@
...
+Many editors support ctags either natively or via a plugin (e.g. vim, emacs,
+BBEdit, ...). For example, to use it in vim, start it in that directory and
+then do:
   :tag DerivedSubgroup
 (say). You can also use
   :ts DerivedSubgroup
 to get an overview over all tags found (which is useful since it shows
 for example all method installations).
+
+For instructions on how to create tags automatically via git-hooks, for example
```
Comment (fingolfin) [inline etc/ctags.readme.txt:42]:
```suggestion
For instructions on how to regenerate tags automatically via git-hooks, for example
```

**Comment by ssiccha** [comment]
I've converted the file into markdown, which also IMO makes it a lot easier to read.

File: etc/ctags.readme.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    :tag <function-name>
+
+or, while your cursor is on the name of that function, press `<CTRL-]>`
+to jump to the first definition of that function. If for example an operation has
+several several methods installed, you can also do
+
+    :ts <function-name>
+
+or press <g]> or to get an overview over all tags found.
```
Comment (ruthhoffmann) [inline etc/ctags.readme.md:63]: What is <g]> ? It's also missing the code backticks formatting.

**Review by ruthhoffmann — COMMENTED** [review]
Except for the <g]> this looks a lot better!

File: etc/ctags.readme.md (hunk 7 above)
Comment (ssiccha) [reply etc/ctags.readme.md]: I've added the backticks, thanks! `<g]>` is a vim keybinding and means "press `g`, then `]`".

Comment (ruthhoffmann) [reply etc/ctags.readme.md]: Oh I see. :D Of course now it makes sense!

**Review by ruthhoffmann — APPROVED** [review]
That looks good to me now!

**Review by fingolfin — APPROVED** [review]

**Pull request merged**